docs/investigations/api-documentation/generate-api-documentation.md

```diff
@@ -0,0 +1,70 @@
+# Generating API documentation
+
+## Overview
+
+This document discusses how to create a static documentation page for our API.
+
+As of right now, there is no static page with our API documentation, so if any problem occurs the user can either check the `examples` directory or read the schema on his own.
+
+## Requirements
+* Static page hosted on Compass
+* Shows how the requests should look like
+* Utilises our example requests
+* The examples are compatible with the Director version
+
+## Possible solutions
+
+### 1. Links to the examples on the API playground
+The GraphQL playground that we use on our endpoints generates documentation with comments from the schema.
+We could create a gqlgen plugin that would add a relative example link to every query and mutation. The example files would be stored inside the Director images and available on a GET request to `/director/examples/{file-name}.graphql`. That way the user would be able to check how the request looks right away.
+
+**Work that has to be done**
+* create the gqlgen plugin
+* host and serve the examples on the director
+* make sure that GET requests on the director require no authentication
+* create an html template to which to insert the example
+
+**Pros**
+
+* relatively small amount of work to be done
+* examples are compatible with the Director version
+* requires almost no maintenance
+
+**Cons**
+* requires some amout of work to make the presentation pretty
```

Comment:
hmm, what do you think about this one:

- requires some amount of work to serve pretty GraphQL examples